lib/extract/netcdf/lfric/kernel_data_netcdf.jinja

```diff
@@ -0,0 +1,249 @@
+!! the requested input- and output-parameters to a NetCDF file.
+!! 
+
+module extract_psy_data_mod
+
+    use, intrinsic :: iso_fortran_env, only : int64, int32
+    use extract_netcdf_base_mod, only: ExtractNetcdfBaseType, CheckError
+
+    implicit none
+
+    !> This is the data type that manages the information required
+    !! to write data to a NetCDF file using the PSyData API. A
+    !! static instance of this type is created for each instrumented
+    !! region with PSyclone (and each region will write a separate
+    !! file).
+    type, extends(ExtractNetcdfBaseType), public:: extract_PsyDataType
+ 
+    contains
+        ! The various procedures defined here
+        procedure :: DeclareFieldDouble,       WriteFieldDouble
+        procedure :: DeclareFieldVectorDouble, WriteFieldVectorDouble
+
+        {# Collect the various procedures for the same generic interface #}
+        {# ------------------------------------------------------------- #}
+        {% set all_declares=[] %}
+        {% set all_writes=[] %}
+        {% set all_reads=[] %}
+        {% for name, type, bits in ALL_TYPES %}
+          {{ all_declares.append("DeclareScalar"~name) or "" -}}
+          {{ all_writes.append("WriteScalar"~name) or "" -}}
+          {{ all_reads.append("ReadScalar"~name) or "" -}}
+          {% for dim in ALL_DIMS %}
+            {{ all_declares.append("DeclareArray"~dim~"d"~name) or "" -}}
+            {{ all_writes.append("WriteArray"~dim~"d"~name) or "" -}}
+            {{ all_reads.append("ReadArray"~dim~"d"~name) or "" -}}
+          {% endfor %}
+        {% endfor %}
+
+        {% set indent="            " %}
+        !> Declare generic interface for PreDeclareVariable:
+        generic, public :: PreDeclareVariable => &
+            DeclareFieldDouble, &
+            DeclareFieldVectorDouble, &
+            {{all_declares|join(", &\n"+indent) }}
+
+        !> The generic interface for providing the value of variables,
+        !! which in case of the read-only verification either computes
+        !! the checksum (before a kernel), or compares a checksum (after
+        !! a kernel call).
+        generic, public :: ProvideVariable => &
+            WriteFieldDouble,       &
+            WriteFieldVectorDouble, &
+            {{all_writes|join(", &\n"+indent) }}
+
+        !> The generic interface for reading in variables previously
+        !! written. This is not part of the standard PSyData API, but
+        !! it is convenient for a driver that e.g. reads previously written
+        !! files. The driver does not use LFRic fields (only Fortran arrays),
+        !! so there is no need to provided field-specific read functions.
+        generic, public :: ReadVariable => &
+            {{all_reads|join(", &\n"+indent) }}
+
+    end type extract_PSyDataType
+
+Contains
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares a double precision LFRic field.
```

Comment:
I've fixed that properly, stupid of me to hardcode that, since we already have a generic interface. So I've removed all references to 'double' from the field and vector field functions.